File: backend/model/Group.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const groupSchema = new Schema({
    title: {
        type: String,
        required: true
    },

    subject: {
        type: String,
        required: true
    },

    resource_ids: [{
        type: Schema.Types.ObjectId,
        ref: 'Resource'
    }],

    task_ids: [{
        type: Schema.Types.ObjectId,
        ref: 'Task'
    }],

    created_at: {
        type: Date,
        default: Date.now()
    },

    created_by: {
        type: Schema.Types.ObjectId,
        ref: 'User'
    },

    user_ids: [{
        type: Schema.Types.ObjectId,
        ref: 'User',
    }],

    invited_user_ids: [{
        type: Schema.Types.ObjectId,
        ref: 'User',
    }]
});

module.exports = mongoose.model('Group', groupSchema);

File: AdminPanel/DataAccess/Entities/Group.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace StudyBuddySAP.DataAccess.Entities

{

    public class Group

    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public ObjectId _id { get; set; }
        [BsonElement("title", Order = 2)]
        public string title { get; set; }
        [BsonElement("subject", Order = 3)]
        public string subject { get; set; }
        [BsonElement("resource_ids", Order = 4)]
        public ObjectId[]? resource_ids { get; set; }
        [BsonElement("task_ids", Order = 5)]
        public ObjectId[]? task_ids { get; set; }
        [BsonElement("created_at", Order = 6)]
        public DateTime created_at { get; set; }
        [BsonElement("created_by", Order = 7)]
        public ObjectId created_by { get; set; }
        [BsonElement("user_ids", Order = 8)]
        public ObjectId[]? user_ids { get; set; }
        [BsonElement("invited_user_ids", Order = 9)]
        public ObjectId[]? invited_user_ids { get; set; }
        [BsonElement("__v", Order = 10)]
        public int __v { get; set; }

    }

}